src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+			
+			@Override
+			public void setSelectedValuesList(List<Object> selectedValuesList, int[] selectedIndices) {
+				selectedJudgingTypeIndexes = selectedIndices;
+				updateNosLabel(selectedValuesList);
+			}
+		});
+		selectNoSolutionsFrame.setVisible(true);
+	}
+
+	private SubmissionSampleLocation[] toArray(SubmissionSolutionList list) {
+		return (SubmissionSampleLocation[]) list.toArray(new SubmissionSampleLocation[list.size()]);
+	}
+	
+	/**
+	 * Update selected languages label and selected language list
+	 * @param valuesList
+	 */
+	protected void updateLanguagesLabel(List<Object> valuesList) {
+	    if (valuesList == null || valuesList.size() == 0) {
+	        selectedLanguagesLabel.setText("None selected");
+	        selectedLanguagesLabel.setToolTipText("None selected");
+        } else {
+            selectedLanguagesLabel.setText(valuesList.size()+" selected");
+            selectedLanguagesLabel.setToolTipText(Arrays.toString(valuesList.toArray()));
+            
+            selectedLanguageList = new ArrayList<Language>();
+            valuesList.forEach((lang) -> {selectedLanguageList.add((Language)lang);});
+        }
+	}
+
+	protected void updateProblemsLabel(List<Object> valuesList) {
+	    
+	    // TODO write code
+	    if (valuesList == null || valuesList.size() == 0) {
+            selectedProblemsLabel.setText("None selected");
+            selectedProblemsLabel.setToolTipText("None selected");
+        } else {
+            
+            selectedProblemsLabel.setText(valuesList.size()+" selected");
+            selectedProblemsLabel.setToolTipText(Arrays.toString(valuesList.toArray()));
+            
+            selectedProblemList = new ArrayList<Problem>();
+            valuesList.forEach((prob) -> {selectedProblemList.add((Problem)prob);});
+        }
+	}
+
+	protected void updateNosLabel(List<Object> valuesList) {
+
+        if (valuesList == null || valuesList.size() == 0) {
+            selectedNosLabel.setText("None selected");
+            selectedNosLabel.setToolTipText("None selected");
+        } else {
+
+            selectedNosLabel.setText(valuesList.size() + " selected");
+            selectedNosLabel.setToolTipText(Arrays.toString(valuesList.toArray()));
+            submissionSolutionList = new SubmissionSolutionList();
+
+            // update/load selected items into submissionSolutionList
+            valuesList.forEach((subSL) -> {
+                submissionSolutionList.add((SubmissionSampleLocation) subSL);
+            });
+
+        }
+    }
+
+	protected void clearTextArea() {
+		textArea.selectAll();
+		textArea.replaceSelection("");
+		linenumber = 0;
+	}
+
+	void addLineToTextArea(String s) {
+
+		linenumber++;
+		String fmtLineNumber = String.format("%4d", linenumber);
+
+		textArea.insert(fmtLineNumber + " " + hhMMSSformatter.format(new Date()) + " " + s + "\n", 0);
+//		textArea.append(s);
+	}
+
+	protected void selectNewCDP() {
```

Comment:
The purpose of this method appears to be to provide the user with the ability to select a new location containing a CDP (which is logical).  However, the first thing it does (below) is call method `selectEntry()`.  That method _forces_ the application of a `FileNameExtensionFilter` which restricts the displayed choices to "files ending in `.yaml`".   This makes it impossible to select a CDP _directory_ -- which is what the user is highly likely to be trying to do.  (The label on the textfield component holding the CDP name is even `CDP config dir`, making it obvious that this method (which is the actionListener for the button used to select the CDP) should be allowing the user to select a _directory_.  

At present, the only way to select a different CDP is to understand that one must actually _select a `.yaml` contained within the CDP directory.  The average user is _highly unlikely_ to be able to intuit this from the GUI.   The label on the text field is `CDP config dir`; that's what the "select" button should allow the user to select.  (And no, simply changing the _label_ isn't an acceptable solution; it'll still leave the user confused about how to select the CDP directory).

I consider this an approval showstopper -- something which needs to be fixed before I'd be willing to approve the PR.